### PR TITLE
gtk4: Add sample/examples/application8

### PR DESCRIPTION
**Pull request by ToshioCP**
Add gtk4/sample/examples/application8

### CODE REVIEW

**Comment by ToshioCP** [comment]
The script `exampleapp.rb` uses ruby methods instead of C functions.

```ruby
291:     strings = buffer.text.downcase.split(/\W+/).uniq
292:     strings.delete("")
```

It can be modified to use C functions like this:

```ruby
    strings = []
    start_iter = buffer.start_iter
    until start_iter.end?
      until start_iter.starts_word || start_iter.end?
        start_iter.forward_char
      end
      break if start_iter.end?
      end_iter = start_iter.clone
      end_iter.forward_word_end
      word = buffer.get_text(start_iter, end_iter, false).downcase
      strings << word
      start_iter = end_iter
    end
    strings = strings.uniq
```

If you prefer the code above, I'll modify it.

File: gtk4/sample/examples/application8/exampleapp.rb [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+    @settings = Gio::Settings.new("org.gtk.exampleapp")
+    @settings.bind("transition", stack, "transition-type", :default)
+    @settings.bind("show-words", sidebar, "reveal-child", :default)
+    search.bind_property("active", searchbar, "search-mode-enabled", :bidirectional)
+    sidebar.signal_connect "notify::reveal-child" do |_sidebar, _pspec|
+      update_words
+    end
+    action = @settings.create_action("show-words")
+    add_action(action)
+end
```
Comment (kou) [inline gtk4/sample/examples/application8/exampleapp.rb:249]:
```suggestion
  end
```

File: gtk4/sample/examples/application8/exampleapp.rb [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+    buffer.select_range(range[0], range[1])
+    view.scroll_to_iter(range[0], 0.0, false, 0.0, 0.0)
+  end
+
+  def update_words
+    tab = stack.visible_child
+    return unless tab
+    view = tab.child
+    buffer = view.buffer
+    strings = buffer.text.downcase.split(/\W+/).uniq
```
Comment (kou) [inline gtk4/sample/examples/application8/exampleapp.rb:291]: `\W+` doesn't work with non-ASCII characters such as Japanese word. How about using the default separator?

```suggestion
    strings = buffer.text.downcase.split.uniq
```

**Review by kou — COMMENTED** [review]
Let's use Ruby implementation.
Ruby implementation is easy to read and calling many C functions is slow.

File: gtk4/sample/examples/application8/exampleapp.rb (hunk 2 above)
Comment (ToshioCP) [reply gtk4/sample/examples/application8/exampleapp.rb:291]: The original program picks up "words" which are defined in Pango.
And Pango supports Unicode.
Words don't include ";",  ":" or some other characters.
The default argument of the split method is `$;` and `$;` is usually nil.
Then split uses space characters as delimiters, which don't include ";" and ":".
Therefore, the behavior is different.
For example,

Sample text:
abc def; ghi.

Original program:
=> ["abc", "def", "ghi"]

Split method with the default argument:
=> ["abc", "def;", "ghi."]

I think maybe the following code works in Unicode texts.

```ruby
strings = buffer.text.downcase.split(/[^[:word:]]+/).uniq
```

How about it?

Comment (ToshioCP) [reply gtk4/sample/examples/application8/exampleapp.rb:291]: I found that `example.app` didn't support Japanese text.
Maybe we need to modify:

`stream.read` => `stream.read.force_encoding(Encoding::UTF_8)`

Comment (kou) [reply gtk4/sample/examples/application8/exampleapp.rb:291]: `strings = buffer.text.downcase.scan(/[[:word:]]+/).uniq` is better. We can remove `strings.delete("")` with it.

```ruby
file.read do |stream|
  buffer.text = stream.read.force_encoding(Encoding::UTF_8)
end
```

is better.

**Comment by ToshioCP** [comment]
I've changed exampleapp.rb to support UTF-8 encoding.
In addition,aplication3/exampleapp.rb to application7/exampleapp.rb are also fixed.

**Comment by kou** [comment]
Thanks!

**Pull request merged**